jellyfin-api/src/main/kotlin/org/jellyfin/apiclient/api/client/KtorClient.kt

```diff
@@ -89,8 +90,17 @@ public open class KtorClient(
 			// Append query parameters
 			queryParameters
 				.filterNot { it.value == null }
-				.forEach {
-					parameters.append(it.key, it.value.toString())
+				.forEach { (key, rawValue) ->
+					// Determine value
+					val value = when (rawValue) {
+						// Lists should be comma-separated
+						is Collection<*> -> if (rawValue.isNotEmpty()) rawValue.joinToString(",") else null
+						else -> rawValue.toString()
+					}
+
+					// Add not-null values
+					if (value != null)
+						parameters.append(key, value)
```

Comment:
Instead of the null-check, you could do `return@forEach` instead. Reads a bit neater imo.

Reply:
I'd say that this is more readable. A return is easy to miss when reading the code while this indent clearly shows the branching.

Reply:
I'm not a fan of the extra null-check, but it's a matter of taste I guess ^^